project/npda/views/home.py

```diff
@@ -51,20 +56,37 @@ def home(request):
     """
     if request.method == "POST":
         form = UploadFileForm(request.POST, request.FILES)
-        file = request.FILES["csv_upload"]
+        user_csv = request.FILES["csv_upload"]
         pz_code = request.session.get("pz_code")
 
-        # summary = csv_summarize(csv_file=file)
+        # You can't read the same file twice without resetting it
+        user_csv.seek(0)
```

Comment:
I wonder if it's easier to pull out reading the CSV file into a dataframe to this level and then pass the dataframe to each function (summarise, check_headers and finally csv_upload?). That way we don't need to remember to seek the file and we guarantee that there's no difference between how Pandas parses the CSV and how the stdlib `csv` module does.

Reply:
good idea: possibly needs some simple validation step here instead then that it is actually a csv with the correct headings and not auntie's investments or the doctor rota uploaded by mistake. If we are not rejecting it, we could store in a dataframe and store the csv in `media` at that point, then run the validation and save steps at that point.

Reply:
@dc2007git are you ok for me to take over this PR and finish it off? There's a few more related issues that came up in the last NPDA catch up which I'd like to fold into this PR: https://github.com/rcpch/national-paediatric-diabetes-audit/issues/348, https://github.com/rcpch/national-paediatric-diabetes-audit/issues/349